packages/vite/src/node/importGlob.ts

```diff
@@ -79,8 +86,12 @@ export async function transformImportGlob(
       ;[importee] = await normalizeUrl(file, pos)
     }
     imports.push(importee)
-    const identifier = `__glob_${importIndex}_${i}`
-    if (isEager) {
+    if (isRaw) {
+      entries += ` ${JSON.stringify(file)}: ${JSON.stringify(
+        await fsp.readFile(path.join(base, file), 'utf-8')
+      )},`
```

Comment:
I'm curious if we want to handle `?raw` specifically? What if we run with a different approach for this PR:

1. Run `cleanUrl` on the pattern, but also keep track of the `?` suffix in a variable.
2. Run `glob.sync` on this clean pattern
3. When rebuilding the imports, add back the `?` suffix

I think this should then support all suffix as it works generically.

Reply:
The current solution refers to `assertPlugin`. But you are right, deleting the `?` directly cannot solve the following case:
`import.meta.globEager('../assets/svgIcons/file-?.svg?raw')`

I think it would be better to only delete `?raw`  in the pattern.

Reply:
Ah yeah I forgot `?` is valid for globbing too. Maybe we can enforce with `\?`? Might be a bit weird but explicit. 

The reason I prefer a more generic approach is so that it would work with image preprocessing that uses suffixes too, eg `vite-imagetools`. I think it's one of the common usecases.

Reply:
I think just use `?raw`  is OK, it's a bit redundant to use `\?raw`

Reply:
I'm still a bit wary of not doing `\?raw` since that adds Vite-specific logic for globbing, which makes it a bit non-standard, but that's my opinion :smile: I'll see if anyone else has an opinion on this.

Reply:
We discussed with @sodatea and @Shinigami92 about this, and we think that we should follow import assertions here:
```js
import.meta.glob("./foo-?.png",{ assert: { type: "raw" } })
```
See https://github.com/tc39/proposal-import-assertions#synopsis
We still need to discuss this with Evan, but at least to me, even if it is a little bit verbose it is good to align ourselves with the standard.

Reply:
Resolving this one, let's use the main thread to discuss